go/tools/internal/stdlib_tags/stdlib_tags.go

```diff
@@ -0,0 +1,161 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"go/build/constraint"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+// Used to update the list of tags affecting the standard library kept in
+// transitions.bzl.
+func main() {
+	if len(os.Args) < 2 {
+		println("usage: stdlib_tags <go SDK src directory>...")
+		os.Exit(1)
+	}
```

Comment:
optional nit: Generally prefer to only exit once in `main()`, and also prefer to parameterize the Output as a writer.

This way, we can write a test for this function with some other `io.Writer`

```suggestion
// Used to update the list of tags affecting the standard library kept in
// transitions.bzl.
func main() {
	if err := run(os.Stdout); err != nil {
		log.Fatal(err)
	}
}
func run(output io.Writer) error {
	if len(os.Args) < 2 {
		return errors.New("usage: stdlib_tags <go SDK src directory>...")
	}
	...
	_, err := fmt.Fprintf(output, "_TAG_AFFECTS_STDLIB = {\n")
	...
```

Reply:
I refactored out a testable function, but left in the prints as we would have to distinguish between `stdout` and `stderr` and the remainder of the `main` function can't really be effectively unittested anyway.